### DISCUSSION TITLE
Bug

### DISCUSSION THREAD

**Discussion by kuzhijie** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1342] [Q&A]
![image](https://github.com/user-attachments/assets/3c24ad51-fabf-4eb2-8668-5ed291d5e885)

In the latest version 1.28.4,  the Charge Limiter doesn't work well, it often charge the battery to full capacity and Automatic Discharge to less than 15%. I am sure that I didn't active the Calibration Mode.    I don't know why it is charged over the limit battery percentages I set.

**Comment by MatthiasKerbl** [comment]
Hi @kuzhijie,

I am sorry to hear you are experiencing issues with AlDente. If possible, please share a debug file the next time this happens so that we can assess this issue better. You can find a guide on [How to generate and share a debug file](https://apphousekitchen.com/how-to-generate-and-share-a-debug-file/) on our blog.

Best,
Matthias

**Reply by kuzhijie** [reply]
![image](https://github.com/user-attachments/assets/9accde0b-d90b-4de7-8478-ece92ca5b10d)

[aldentepro-debug-2024-10-01-07:19:58.pasta.zip](https://github.com/user-attachments/files/17197405/aldentepro-debug-2024-10-01-07.19.58.pasta.zip)

Hello Matthias,

Thank you for your reply. In an effort to protect my laptop battery, I reverted AlDente to a previous version (1.26.1) .

Just a moment ago, I noticed that my battery began discharging to a low percentage again, despite me not taking any action. I've attached the log file for your review. Could you please check it and advise?

Thank you for your assistance.

**Reply by MatthiasKerbl** [reply]
Hi @kuzhijie,

Thank you for sharing the debug file.
It looks like your Battery Percentage is jumping around quite a bit.

- Is it possible you have a replacement battery installed?
- When was the last time you fully charged and discharged your MacBook or used the Calibration Mode?

Best,
Matthias

**Reply by kuzhijie** [reply]
![image](https://github.com/user-attachments/assets/36cfbbf8-93a4-4565-bb4f-479145472ce8)

Hello Matthias,
  

1. I replaced my battery 40 days ago, which is actually why I started using AIDente Pro. My goal is to prolong the lifespan of this new battery, as the replacement was quite costly.
2. I've set up a Calibration Mode schedule that runs once a week. All records of these calibrations are included in the screenshot I've attached.

Additionally, I wanted to bring your attention to an unusual phenomenon I observed recently. While the charge limit was set to 50% and had been maintaining that level, the battery power suddenly dropped to 12% when the Calibration Mode started this time.  I've attached the log file for your reference. 

Thank you very much. 

[aldentepro-debug-2024-10-03-22:40:29.pasta.zip](https://github.com/user-attachments/files/17245893/aldentepro-debug-2024-10-03-22.40.29.pasta.zip)

**Reply by MatthiasKerbl** [reply]
Hi @kuzhijie,

Did you get your battery replaced at Apple directly?
It looks like something is off with the replacement battery which results in the battery percentage being highly inaccurate and "jumping around".

Best,
Matthias

**Reply by kuzhijie** [reply]
Hello Matthias,

My computer is out of warranty, and the official battery repair is expensive, so I bought a third-party battery. Is there a way to solve this problem?

**Reply by MatthiasKerbl** [reply]
Hi @kuzhijie,

Not from our side, unfortunately. However, you can try to get a replacement or refund for the third-party battery.

Best,
Matthias
